### PR TITLE
discovery: put service name generated from container tags in a separate field

### PR DESCRIPTION
**Pull request by Yumasi**
…te field

### What does this PR do?

This PR adds two fields to the Service Discovery payload for the service name generated from container tags introduced in #31306:
- `ContainerServiceName`: to prevent overriding the service name generated by looking at process information.
- `ContainerServiceNameSource`: the name of the tag that was used to generate the service name.

### Describe how you validated your changes

Covered by automatic tests.

### CODE REVIEW

File: pkg/collector/corechecks/servicediscovery/events.go [hunk 1]
```diff
@@ -30,6 +30,7 @@ type eventPayload struct {
 	ServiceName                string   `json:"service_name"`
 	GeneratedServiceName       string   `json:"generated_service_name"`
 	GeneratedServiceNameSource string   `json:"generated_service_name_source,omitempty"`
+	ContainerServiceName       string   `json:"container_service_name"`
```
Comment (guyarb) [inline pkg/collector/corechecks/servicediscovery/events.go:33]: should the source tag be added as well as a new field?

File: pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go [hunk 2]
```diff
@@ -847,9 +847,10 @@ func TestDocker(t *testing.T) {
 	require.Contains(t, portMap, pid1111)
 	require.Contains(t, portMap[pid1111].Ports, uint16(1234))
 	require.Contains(t, portMap[pid1111].ContainerID, "dummyCID")
-	require.Contains(t, portMap[pid1111].Name, "foo_from_app_tag")
-	require.Contains(t, portMap[pid1111].GeneratedName, "foo_from_app_tag")
-	require.Contains(t, portMap[pid1111].GeneratedNameSource, string(usm.Container))
```
Comment (vitkyrka) [inline pkg/collector/corechecks/servicediscovery/module/impl_linux_test.go:852]: Perhaps remove this `Container` enum value?

File: pkg/collector/corechecks/servicediscovery/module/impl_linux.go [hunk 3]
```diff
@@ -686,22 +687,18 @@ func (s *discovery) enrichContainerData(service *model.Service, containers map[s
 	serviceName := getServiceNameFromContainerTags(container.Tags)
 
 	if serviceName != "" {
-		service.GeneratedName = serviceName
+		service.ContainerServiceName = serviceName
 		// Update the legacy name field as well
-		if service.DDService == "" {
+		if service.Name == "" {
```
Comment (vitkyrka) [inline pkg/collector/corechecks/servicediscovery/module/impl_linux.go:692]: Is this code for setting `service.Name` needed now? Maybe we should just leave that to be ddservice/generated and not involve the container name there?

File: pkg/collector/corechecks/servicediscovery/events.go (hunk 1 above)
Comment (vitkyrka) [inline pkg/collector/corechecks/servicediscovery/events.go:33]: `omitempty`?

**Review by vitkyrka — APPROVED** [review]

File: pkg/collector/corechecks/servicediscovery/events.go (hunk 1 above)
Comment (Yumasi) [reply pkg/collector/corechecks/servicediscovery/events.go]: Added in 3aaed1d765eb991528e8c0d3f42278a085aaa3a4

**Pull request merged**